workspace-manager-clienttests/src/main/java/scripts/uploadscripts/SummariesToBigQuery.java

```diff
@@ -0,0 +1,215 @@
+        rowContent.put(
+                "elapsedTime_standardDeviation", testScriptResult.elapsedTimeStatistics.standardDeviation);
+        rowContent.put("elapsedTime_median", testScriptResult.elapsedTimeStatistics.median);
+        rowContent.put("elapsedTime_percentile95", testScriptResult.elapsedTimeStatistics.percentile95);
+        rowContent.put("elapsedTime_percentile99", testScriptResult.elapsedTimeStatistics.percentile99);
+        rowContent.put("elapsedTime_sum", testScriptResult.elapsedTimeStatistics.sum);
+
+        rowContent.put("totalRun", testScriptResult.totalRun);
+        rowContent.put("numCompleted", testScriptResult.numCompleted);
+        rowContent.put("numExceptionsThrown", testScriptResult.numExceptionsThrown);
+        rowContent.put("isFailure", testScriptResult.isFailure);
+
+        return rowContent;
+    }
+
+    /** Build a single row for each test run. */
+    private Map<String, Object> buildTestRunRow(Path outputDirectory) throws JsonProcessingException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        Map<String, Object> rowContent = new HashMap<>();
+
+        rowContent.put("id", testRunSummary.id);
+        rowContent.put("testConfig_name", renderedTestConfiguration.name);
+        rowContent.put("testConfig_description", renderedTestConfiguration.description);
+        rowContent.put("server_name", renderedTestConfiguration.server.name);
+        rowContent.put(
+                "kubernetes_numberOfInitialPods", renderedTestConfiguration.kubernetes.numberOfInitialPods);
+        rowContent.put(
+                "testUsers", renderedTestConfiguration.testUsers.stream().map(tu -> tu.name).toArray());
+
+        TimeZone originalDefaultTimeZone = TimeZone.getDefault();
+        TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
+        rowContent.put(
+                "startTime", Timestamp.from(Instant.ofEpochMilli(testRunSummary.startTime)).toString());
+        rowContent.put(
+                "startUserJourneyTime",
+                Timestamp.from(Instant.ofEpochMilli(testRunSummary.startUserJourneyTime)).toString());
+        rowContent.put(
+                "endUserJourneyTime",
+                Timestamp.from(Instant.ofEpochMilli(testRunSummary.endUserJourneyTime)).toString());
+        rowContent.put(
+                "endTime", Timestamp.from(Instant.ofEpochMilli(testRunSummary.endTime)).toString());
+        TimeZone.setDefault(originalDefaultTimeZone);
+
+        rowContent.put("outputDirectory", outputDirectory.toAbsolutePath().toString());
+        rowContent.put(
+                "json_testConfiguration", objectMapper.writeValueAsString(renderedTestConfiguration));
+        rowContent.put("json_testRun", objectMapper.writeValueAsString(testRunSummary));
+        rowContent.put(
+                "json_measurementCollection",
+                objectMapper.writeValueAsString(measurementCollectionSummaries));
+
+        return rowContent;
+    }
```

Comment:
if TestRunner has a `BigQueryUtils`, and TestRunner defines the data structure for the test script specifications and test result summaries, could `buildTestScriptResultsRow` and `buildTestRunRow` move into TestRunner instead of living here?

Reply:
That's exactly the plan as stated in my previous response.